test/battle/move_effect/skill_swap.c

```diff
@@ -0,0 +1,109 @@
+#include "global.h"
+#include "test/battle.h"
+
+ASSUMPTIONS
+{
+    ASSUME(gMovesInfo[MOVE_SKILL_SWAP].effect == EFFECT_SKILL_SWAP);
+}
+
+SINGLE_BATTLE_TEST("Skill Swap swaps user and target's abilities")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) { Ability(ABILITY_TELEPATHY); }
+        OPPONENT(SPECIES_CHARMANDER) { Ability(ABILITY_BLAZE); }
+    }WHEN {
+        TURN { MOVE(player, MOVE_SKILL_SWAP); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_SKILL_SWAP, player);
+        ABILITY_POPUP(player, ABILITY_TELEPATHY);
+        ABILITY_POPUP(opponent, ABILITY_BLAZE);
+    } THEN {
+        EXPECT_EQ(player->ability, ABILITY_BLAZE);
+        EXPECT_EQ(opponent->ability, ABILITY_TELEPATHY);
+    } 
+}
+
+DOUBLE_BATTLE_TEST("Skill Swap only swaps user's ability with target's ability")
```

Comment:
I assume this test checks if partners still have their original ability? If that is the case then it should be checked in `THEN`